Content/Application/PropertyResolver/Resolver/BlockPropertyResolver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContentBundle\Content\Application\PropertyResolver\Resolver;
+
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FieldMetadata;
+use Sulu\Bundle\ContentBundle\Content\Application\ContentResolver\Value\ContentView;
+use Sulu\Bundle\ContentBundle\Content\Application\MetadataResolver\MetadataResolver;
+use Sulu\Bundle\ContentBundle\Content\Application\PropertyResolver\PropertyResolverInterface;
+
+class BlockPropertyResolver implements PropertyResolverInterface
+{
+    private MetadataResolver $metadataResolver;
+
+    /**
+     * Prevent circular dependency by injecting the MetadataResolver after instantiation.
+     */
+    public function setMetadataResolver(MetadataResolver $metadataResolver): void
```

Comment:
```suggestion
    /**
     * @internal 
     *
     * Prevent circular dependency by injecting the MetadataResolver after instantiation.
     */
    public function setMetadataResolver(MetadataResolver $metadataResolver): void
```

This method should not be part of the public API.